src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -241,6 +241,15 @@ internal class NativeCommandProcessor : CommandProcessorBase
             ".vbs",
         };
 
+        // This is the list of PowerShell names.
+        // It's used by the MiniShell to determine if a command is a PowerShell.
+        private static readonly IReadOnlySet<string> s_powerShellExecutableNames = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+        {
+            "powershell.exe",
+            "pwsh",
+            "pwsh.exe",
```

Comment:
`gsudo` needs to be added to this list.  See https://twitter.com/gerardo_gr/status/1582934397820751873

Which means this list should be renamed to something else like `s_commandsSupportingEncodedCommand`

Reply:
Don't forget about me!

Also, to make things worse, here is another interesting edge case. You can `sudo pwsh { scriptblock }` is currently working well on linux.

![image](https://user-images.githubusercontent.com/3901474/200612683-f179152e-04e9-4803-a9b8-e7f8dcc3b272.png)